### PR TITLE
Feat/restaurant middleware

### PR DESCRIPTION
**Pull request by jx0906**
unable to test FE due to all the existing validation workflows (ie, checking of jwt, only displaying of rest info for users with owner roles) so tested the code by attempting to directly edit rest data from BE (through Postman).

In the screenshot below, I had attempted to (i) update and (ii) delete restaurant info for an existing restaurant entry tagged with owner info. Got the feedback msg that its "unauthorised" (expected behaviour).

**original restaurant entry with owner info:**
image

**(i) Update info -- > got feedback of unauthorised operation - because I'm not logged on as an owner in Postman:**
image

**(ii) Delete info -- > got feedback of unauthorised operation - because I'm not logged on as an owner in Postman:**
image

### CODE REVIEW

**Comment by jx0906** [comment]
FE experience as follows (for a legitimate workflow):

**_User's restaurant info_**

image

**_Edit rest info as owner_**

image

image

**_Delete rest info as owner_**

image

_Note - backend returns no results for search of restaurant name_
 
image

**_User can now create another restaurant as he is still holding an Owner account_**

image

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,6 @@ node_modules/
 .DS_Store
 .vscode/*
 !.vscode/extensions.json
-.env
\ No newline at end of file
+.env
+package-lock.json
+package.json
```
Comment (natsumi-h) [inline .gitignore:7]: `package-lock.json`
`package.json` should not be ignored.

**Review by natsumi-h — CHANGES_REQUESTED** [review]
* If you want to work on the old branch with additional feature, merge main branch to your working branch before pushing additional work. Otherwise it causes unexpected rollback.
* `package-lock.json`
`package.json` should not be ignored.

**Comment by jx0906** [comment]
pardon, didn't know the old branches will not be merged with the changes from main even with git pull. will close this branch and create a new one for the code changes!

**Pull request closed without merging**